Sources/CacheAdvance/FileHandleExtensions.swift

```diff
@@ -23,7 +23,13 @@ extension FileHandle {
 
     /// A method to read data from a file handle that is safe to call in Swift from any operation system version.
     func readDataUp(toLength length: Int) throws -> Data {
-        if #available(iOS 13.0, tvOS 13.0, watchOS 6.0, macOS 10.15, *) {
+        if #available(iOS 13.4, tvOS 13.4, watchOS 6.2, macOS 10.15.4, *) {
+            if let data = try read(upToCount: length) {
+                return data
+            } else {
+                return Data()
```

Comment:
It's interesting that the method both throws and returns a nullable result. I wonder how you could end up with `nil`? In any case, do you think we should throw our own error if the data is `nil`?

Reply:
Oh we end up with `nil` when the data is empty. I tried throwing here instead of returning `Data()` and I broke a _ton_ of tests. Returning `Data()` here preserves existing behavior.